pkg/apis/core/v1/helper/helpers.go

```diff
@@ -436,3 +438,15 @@ func StorageNodeAffinityToAlphaAnnotation(annotations map[string]string, affinit
 	annotations[v1.AlphaStorageNodeAffinityAnnotation] = string(json)
 	return nil
 }
+
+// IsCriticalPodBasedOnPriority checks if the given pod is a critical pod based on priority resolved from pod Spec.
+func IsCriticalPodBasedOnPriority(ns string, priority int32) bool {
```

Comment:
This is only called from one spot. Move it to the point of use.

Reply:
Agree. SystemCriticalPriority belongs here or the helper into the scheduler.